tools/crop_image/crop_image.xml

```diff
@@ -0,0 +1,105 @@
+    </requirements> 
+    <command detect_errors="aggressive"><![CDATA[
+
+        mkdir ./output &&
+        python '$__tool_directory__/crop_image.py'
+
+        '$image'
+        '$labelmap'
+        '$skip_labels'
+        '${image.ext}'
+
+        ./output
+
+    ]]></command>
+    <inputs>
+        <param name="image" type="data" format="png,tiff" label="Image file" help="The image to be cropped."/>
+        <param name="labelmap" type="data" format="png,tiff" label="Label map" help="Each label identifies an individual region of interest, for which a cropped image is produced."/>
+        <param name="skip_labels" type="text" label="Skip labels" value="0" optional="true" help="Comma-separated list of labels for which no cropped image shall be produced.">
+            <validator type="regex">^\d+(,\d+)*$|^$</validator>
+        </param>
+    </inputs>
+    <outputs>
+        <collection type="list" name="output" label="Crop ${on_string}" format_source="image">
+            <discover_datasets directory="output" pattern="__name_and_ext__"/>
+        </collection>
+    </outputs>
+    <tests>
+        <!-- Test 2D TIFF -->
+        <test>
+            <param name="image" value="yx_float32.tiff" ftype="tiff"/>
+            <param name="labelmap" value="yx_uint8.tiff"/>
+            <output_collection name="output" type="list" count="2">
+                <expand macro="tests/intensity_image_diff/element" name="1" value="yx_float32_uint8_1.tiff" ftype="tiff"/>
+                <expand macro="tests/intensity_image_diff/element" name="2" value="yx_float32_uint8_2.tiff" ftype="tiff"/>
+            </output_collection>
+        </test>
+        <!-- Test with `skip_labels` -->
+        <test>
+            <param name="image" value="yx_float32.tiff" ftype="tiff"/>
+            <param name="labelmap" value="yx_uint8.tiff"/>
+            <param name="skip_labels" value="0,1"/>
+            <output_collection name="output" type="list" count="1">
+                <expand macro="tests/intensity_image_diff/element" name="2" value="yx_float32_uint8_2.tiff" ftype="tiff"/>
+            </output_collection>
+        </test>
+        <!-- Test with empty `skip_labels` -->
+        <test>
+            <param name="image" value="yx_float32.tiff" ftype="tiff"/>
+            <param name="labelmap" value="yx_uint8.tiff"/>
+            <param name="skip_labels" value=""/>
+            <output_collection name="output" type="list" count="3">
+                <expand macro="tests/intensity_image_diff/element" name="0" value="yx_float32_uint8_0.tiff" ftype="tiff"/>
+                <expand macro="tests/intensity_image_diff/element" name="1" value="yx_float32_uint8_1.tiff" ftype="tiff"/>
+                <expand macro="tests/intensity_image_diff/element" name="2" value="yx_float32_uint8_2.tiff" ftype="tiff"/>
+            </output_collection>
+        </test>
+        <!-- Test 3D TIFF (multi-frame) -->
+        <test>
```

Comment:
@kostrykin @bernt-matthias what do you think about a `<test description="Test 3D TIFF (multi-frame)">` thing?

Reply:
I like it, but mainly because XML comments are difficult to comment :) That is, I often want to comment some of the tests and if there is a comment describing the test I need to comment test-by-test.

Reply:
We could re-use this in planemo and  GLS ...

Reply:
+1

Reply:
It would actually be very helpful if the description of the test would be written in the test report (there it currently only says Test 1, Test2, …, Test N, and you have to count by hand to find the corresponding test in the XML).